Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsComponentModeCommon.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/std/string/string.h>
+#include <Editor/EditorJointCommon.h>
+
+namespace PhysX::JointsComponentModeCommon
+{
+    namespace SubComponentModes
+    {
+        enum class ModeType : AZ::u32
+        {
+            Translation = 0,
+            Rotation,
+            MaxForce,
+            MaxTorque,
+            Damping,
+            Stiffness,
+            TwistLimits,
+            SwingLimits,
+            SnapPosition,
+            SnapRotation,
+
+            NumModes
+        };
+
+        struct AngleModesSharedRotationState
+        {
+            AZ::Vector3 m_axis;
+            AZ::Quaternion m_savedOrientation = AZ::Quaternion::CreateIdentity();
+            PhysX::AngleLimitsFloatPair m_valuePair;
+        };
+    } // namespace SubComponentModes
+
+    struct ParamaterNames
+    {
+        static const AZStd::string TwistLimits;
+        static const AZStd::string Damping;
+        static const AZStd::string MaxForce;
+        static const AZStd::string MaxTorque;
+        static const AZStd::string Position;
+        static const AZStd::string Rotation;
+        static const AZStd::string SnapPosition;
+        static const AZStd::string SnapRotation;
+        static const AZStd::string Stiffness;
+        static const AZStd::string SwingLimit;
+        static const AZStd::string Transform;
+        static const AZStd::string ComponentMode;
+        static const AZStd::string LeadEntity;
```

Comment:
suggestion: @jonawals just pointed this out to be in another PR but you can I believe use `static inline const AZStd::string_view Name = "blah";` without having to have them out of line which is just less code to worry about 🙂